[docs] First pass on a local-first guide (#30880)
expo/expo: docs/pages/guides/local-first.mdx

### Prisma

[Prisma](https://prisma.io) is well known as the most popular ORM for Node.js and TypeScript backends, and it's now available for Expo and React Native in early access. [Learn more](https://www.prisma.io/blog/bringing-prisma-orm-to-react-native-and-expo). Prisma aims to provide a complete local-first solution, with state management, syncing, and persistence all covered for you. While it's still early, [Beto Moedano](https://github.com/betomoedano) has put together full walkthrough of using Prisma with Expo to build a local-first Notion clone, [watch it on YouTube](https://www.youtube.com/watch?v=uTrPte0sCiw) and [check out the code on GitHub](https://github.com/betomoedano/React-Native-Notion-Clone).

### Other tools

> Far from a comprehensive list, the following iis a list of some other tools that have caught our attention and that you may find interesting to explore. For a more thorough list of tools on the ["Local-first software" community website](https://localfirstweb.dev/).

Reviewer: ```suggestion
The following list, far from being comprehensive, provides other tools that have caught our attention and that you may find interesting to explore. For a more thorough list of tools, see ["Local-first software" community website](https://localfirstweb.dev/).
```

We can be concise in the first sentence and avoid redundancy when referring the list below.

- [LiveStore](https://github.com/livestorejs/livestore)
- [Automerge](https://automerge.org/)
- [ElectricSQL](https://electric-sql.com/)
- [Instant](https://www.instantdb.com/)
- [PowerSync](https://www.powersync.com/)

## Additional resources

> - ["The past, present, and future of local-first"](https://www.youtube.com/watch?v=NMq0vncHJvU) by Martin Kleppmann
> - ["Local-first software"](https://www.inkandswitch.com/local-first/) by Ink & Switch
> - ["Local-first software" community website](https://localfirstweb.dev/) and [meetup playlist on YouTube](https://www.youtube.com/playlist?list=PLTbD2QA-VMnXFsLbuPGz1H-Najv9MD2-H)

Reviewer: ```suggestion
- [The past, present, and future of local-first](https://www.youtube.com/watch?v=NMq0vncHJvU) by Martin Kleppmann
- [Local-first software](https://www.inkandswitch.com/local-first/) by Ink & Switch
- ["Local-first software" community website](https://localfirstweb.dev/) and [meetup playlist on YouTube](https://www.youtube.com/playlist?list=PLTbD2QA-VMnXFsLbuPGz1H-Najv9MD2-H)
```
